CMakeLists.txt

```diff
@@ -161,11 +128,66 @@ else()
 
   # Only Linux targets need dl for dynamic linking.
   if( CMAKE_SYSTEM_NAME MATCHES "Linux" )
-    target_link_libraries(include-what-you-use
+    list(APPEND IWYU_LINK_LIBS
       dl
     )
   endif()
 endif()
 
+add_library(IWYU_OBJECTS OBJECT
+  iwyu_ast_util.cc
+  iwyu_cache.cc
+  iwyu_driver.cc
+  iwyu_getopt.cc
+  iwyu_globals.cc
+  iwyu_include_picker.cc
+  iwyu_lexer_utils.cc
+  iwyu_location_util.cc
+  iwyu_output.cc
+  iwyu_path_util.cc
+  iwyu_preprocessor.cc
+  iwyu_verrs.cc
+)
+
+add_executable(include-what-you-use
+ iwyu.cc
+ $<TARGET_OBJECTS:IWYU_OBJECTS>
+)
+
+add_executable(include-what-you-use-tests
+  $<TARGET_OBJECTS:IWYU_OBJECTS>
+  more_tests/iwyu_include_picker_test.cc
+  more_tests/iwyu_lexer_utils_test.cc
+  more_tests/iwyu_output_test.cc
+  more_tests/iwyu_string_util_test.cc
+  more_tests/test_main.cc
+  )
+
+if( MSVC )
+  # Disable warnings for IWYU, and disable exceptions in MSVC's STL.
+  add_definitions(
+    -wd4722 # Suppress ''destructor'' : destructor never returns, potential memory leak
+    -D_HAS_EXCEPTIONS=0
+  )
+
+  # Put project in solution folder
+  set_target_properties(include-what-you-use
+    PROPERTIES FOLDER "Clang executables"
+  )
+else()
+  # Disable RTTI, use C++11 to be compatible with LLVM/Clang libraries.
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -std=c++11")
```

Comment:
How can we set `-fno-rtti` only for include-what-you-use? Because it is used by GoogleTest and fails with
```
llvm/utils/unittest/googletest/include/gtest/internal/gtest-port.h:1041:16: error: cannot use typeid with -fno-rtti
  GTEST_CHECK_(typeid(*base) == typeid(Derived))
```